### PR TITLE
Backports 0.16 pr12

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: test/functional/sendheaders.py [hunk 1]
```diff
@@ -232,32 +247,37 @@ def run_test(self):
...
         tip = self.nodes[0].getblockheader(self.nodes[0].generate(1)[0])
         tip_hash = int(tip["hash"], 16)
 
+        inv_node.check_last_announcement(inv=[tip_hash], headers=[])
+        test_node.check_last_announcement(inv=[tip_hash], headers=[])
+
         # TODO this partly fixes the same thing that is fixed by https://github.com/bitcoin/bitcoin/pull/13192
         # This will later conflict when backporting the actual fix. Just take everything from the Bitcoin fix as a
         # resolution
         assert_equal(test_node.check_last_announcement(headers=[], inv=[tip_hash]), True)
```
Comment (UdjinM6) [inline test/functional/sendheaders.py]: This is no longer needed
```suggestion
```

Comment (PastaPastaPasta) [reply test/functional/sendheaders.py]: Why? (13192 is in my backlog and in that commit I removed this)

File: test/functional/test_framework/test_framework.py [hunk 2]
```diff
@@ -464,7 +464,7 @@ def cache_path(n, *paths):
 
             for i in range(MAX_NODES):
                 for entry in os.listdir(cache_path(i)):
-                    if entry not in ['wallets', 'chainstate', 'blocks']:
+                    if entry not in ['wallets', 'chainstate', 'blocks', 'evodb', 'llmq']:
```
Comment (UdjinM6) [inline test/functional/test_framework/test_framework.py]: Ooops, missed one more thing in my previous suggestion
```suggestion
                    if entry not in ['wallets', 'chainstate', 'blocks', 'evodb', 'llmq', 'backups']:
```

File: test/functional/sendheaders.py (hunk 1 above)
Comment (UdjinM6) [reply test/functional/sendheaders.py]: f798771323ace08b2cba8b7877e2e64b273fd03f added `test_node.check_last_announcement(inv=[tip_hash], headers=[])` right above this which does the the same but also pops from `self.last_message` which causes this line to fail now.

**Comment by PastaPastaPasta** [comment]
Resolved comments

**Pull request merged**